File: StudentCredit.Application/Applications/ApplicationsFive/Dtos/SearchApplicationFiveFilter.cs
using StudentCredit.Data.Applications.ApplicationFive.Enums;
using StudentCredit.Data.Common.Enums;

namespace StudentCredit.Application.Applications.ApplicationsFive.Dtos
{
	public class SearchApplicationFiveFilter
	{
		public int? BankId { get; set; }

		public DateTime? From { get; set; }

		public DateTime? To { get; set; }

		public ApplicationFiveType? ApplicationFiveType { get; set; }

		public CommitState? CommitState { get; set; }

		public int Limit { get; set; } = 10;
		public int Offset { get; set; } = 0;
	}
}

File: WebApplication/src/modules/application/application-five/services/search-application-five.filter.ts
import { Injectable } from "@angular/core";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { BaseSearchFilter } from "src/infrastructure/services/base-search.filter";
import { ApplicationFiveType } from "../enums/application-five-type.enum";
import { CommitState } from "../../common/enums/commit-state.enum";

@Injectable()
export class SearchApplicationFiveFilter extends BaseSearchFilter {
  bank: NomenclatureDto;
  bankId: number | null;
  from: Date;
  to: Date;
  applicationFiveType: ApplicationFiveType;
  commitState: CommitState;

  constructor() {
    super(10);
  }
}